engine/new_runtime/src/crypto_utils/crypto_box.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::locked_memory::ProtectedConfiguration;
+use crate::locked_memory::ProtectedMemory;
+use crate::memories::buffer::Buffer;
+use serde::{Deserialize, Serialize};
+use std::{
+    convert::TryFrom,
+    fmt::Debug,
+    hash::{Hash, Hasher},
+    marker::PhantomData,
+};
+
+/// A provider interface between the vault and a crypto box. See libsodium's [secretbox](https://libsodium.gitbook.io/doc/secret-key_cryptography/secretbox) for an example.
+pub trait BoxProvider: 'static + Sized + Ord + PartialOrd {
+    type Error: Debug;
+
+    /// defines the key length for the [`BoxProvider`].
+    fn box_key_len() -> usize;
+    /// defines the size of the Nonce combined with the Ad for the [`BoxProvider`].
+    fn box_overhead() -> usize;
+
```

Comment:
```suggestion
pub trait BoxProvider: 'static + Sized + Ord + PartialOrd {
    type Error: Debug;
    /// defines the key length for the [`BoxProvider`].
    const BOXED_KEY_LEN: usize;
    /// defines the size of the Nonce combined with the Ad for the [`BoxProvider`].
    const BOXED_OVERHEAD: usize;

```
I know this is old code, but couldn't we use [associated consts](https://doc.rust-lang.org/reference/items/associated-items.html#associated-constants) for this?

Reply:
Yeah you could.  They weren't an option when I wrote this.